cmd/influxd/config.go

```diff
@@ -85,11 +88,12 @@ type Config struct {
 	} `toml:"broker"`
 
 	Data struct {
-		Dir                   string   `toml:"dir"`
-		Port                  int      `toml:"port"`
-		RetentionAutoCreate   bool     `toml:"retention-auto-create"`
-		RetentionCheckEnabled bool     `toml:"retention-check-enabled"`
-		RetentionCheckPeriod  Duration `toml:"retention-check-period"`
+		Dir                            string   `toml:"dir"`
+		Port                           int      `toml:"port"`
+		RetentionAutoCreate            bool     `toml:"retention-auto-create"`
+		RetentionCheckEnabled          bool     `toml:"retention-check-enabled"`
+		RetentionCheckPeriod           Duration `toml:"retention-check-period"`
+		ShardGroupPreCreateCheckPeriod Duration `toml:"shard-group-pre-create-check-period"`
```

Comment:
This is a pretty awkwardly-named config variable. :-) How about something as simple as `retention-create-period`?